tests/test_main_api.py

```diff
@@ -0,0 +1,304 @@
+        "/api/v2/users/login",
+        json={
+            "email": register_another_user["email"],
+            "password": valid_password,
+        },
+    )
+    assert response.status_code == 200
+    return response.json()["token"]
+
+@pytest.fixture
+def create_thread(login_user):
+    # Create a thread before each test that requires a thread
+    response = client.post(
+        "/api/v2/threads",
+        headers={
+            "Authorization": f"Bearer {login_user}",
+            "x-mobile-ansari": "ANSARI",
+        },
+    )
+    assert response.status_code == 200
+    return response.json()["thread_id"]
+
+@pytest.mark.asyncio
+async def test_register_new_user():
+    # Test registering a new user with valid credentials
+    base, domain = valid_email_base.split("@")
+    email = f"{base}+{uuid.uuid4()}@{domain}"
+    response = client.post(
+        "/api/v2/users/register",
+        headers={"x-mobile-ansari": "ANSARI"},
+        json={
+            "email": email,
+            "password": valid_password,
+            "first_name": "Jane",
+            "last_name": "Doe",
+        },
+    )
+    assert response.status_code == 200
+
+# This test will fails, because the email is not validated
+# @pytest.mark.asyncio
+# async def test_register_invalid_email():
+#     # Test registering a new user with an invalid email
+#     response = client.post(
+#         "/api/v2/users/register",
+#         headers={"x-mobile-ansari": "ANSARI"},
+#         json={
+#             "email": "invalid_email",
+#             "password": valid_password,
+#             "first_name": "Jane",
+#             "last_name": "Doe",
+#         },
+#     )
+#     assert response.status_code == 400
+#     assert response.json()["detail"] == "Invalid email"
+
+# This will give an error because zxcvbn raises an IndexError when an empty password is given.
+# See this issue https://github.com/dwolfhub/zxcvbn-python/issues/64
+# @pytest.mark.asyncio
+# async def test_register_blank_password():
+#     # Test registering a new user with a blank password
+#     base, domain = valid_email_base.split("@")
+#     email = f"{base}+{uuid.uuid4()}@{domain}"
+#     response = client.post(
+#         "/api/v2/users/register",
+#         headers={"x-mobile-ansari": "ANSARI"},
+#         json={
+#             "email": email,
+#             "password": "",
+#             "first_name": "Jane",
+#             "last_name": "Doe",
+#         },
+#     )
+#     assert response.status_code == 400
+#     assert response.json()["detail"] == "Password is required"
+
+@pytest.mark.asyncio
+async def test_login_valid_credentials(register_user):
+    # Test logging in with valid credentials
+    response = client.post(
+        "/api/v2/users/login",
+        json={
+            "email": register_user["email"],
+            "password": valid_password,
+        },
+    )
+    assert response.status_code == 200
+    assert "token" in response.json()
+
+@pytest.mark.asyncio
```

Comment:
This should be two tests: 

- Login with non-existent account. 
- Login with existent account, but incorrect password.